### PR TITLE
[stable/prometheus-mongodb-exporter] Add envs to deployment template

### PR DESCRIPTION
**Pull request by dashford**
#### What this PR does / why we need it:
This PR allows environment variables to specified as part of the deployment template. The underlying `mongodb-exporter` allows the `mongodb_uri` to be set through an environment variable, this PR will allow that ability.

For our use case, we use Vault to store credentials like these and have a webhook installed to load secrets from Vault at deploy time based on environment variables. This PR would allow us to do that for our deployments.

#### Which issue this PR fixes
None

#### Special notes for your reviewer:
None

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/prometheus-mongodb-exporter/values.yaml [hunk 1]
```diff
@@ -49,6 +49,9 @@ resources: {}
 #   cpu: 100mm
 #   memory: 128Mi
 
+# Extra environment variables that will be passed into the exporter pod
+env: {}
```
Comment (cpanato) [inline stable/prometheus-mongodb-exporter/values.yaml:53]: can you add a test case to test this?

Comment (dashford) [reply stable/prometheus-mongodb-exporter/values.yaml:53]: No problem, I've added one here https://github.com/helm/charts/pull/12615/commits/95b939cfe97ef876a09696c63b37b95810fbd709. I haven't added tests for charts before so let me know if it's alright. Looks ok according to the docs anyway :slightly_smiling_face:

**Comment by steven-sheehy** [comment]
/assign
Since the -mongodb-uri is hardcoded in the command args, did you test that setting the env variable will override that? Just want to make sure since I'm not sure the priority of args vs env.

**Comment by dashford** [comment]
> /assign
> Since the -mongodb-uri is hardcoded in the command args, did you test that setting the env variable will override that? Just want to make sure since I'm not sure the priority of args vs env.

Hey @steven-sheehy, good call - I've added that to [the test](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/helm_charts/12615/pull-charts-e2e/1112675481599086592/) and it looks to be correctly taking the value from `MONGODB_URI` over the command line args:

```
[ ... ]
I0401 11:19:27.122] Args:
I0401 11:19:27.122] -mongodb.uri=mongodb://localhost:9216
[ ... ]
I0401 11:19:27.124] Environment:
I0401 11:19:27.124] MONGODB_URI: mongodb://hostfromenv:9216
[ ... ]
I0401 11:19:28.114] ------------------------------------------------------------------------------------------------------------------------
I0401 11:19:28.115] ==> Logs of container prometheus-mongodb-exporter-lq2niqpo2w-566f59cdcb-prdnp
I0401 11:19:28.116] ------------------------------------------------------------------------------------------------------------------------
I0401 11:19:28.652] time="2019-04-01T11:18:59Z" level=error msg="Cannot connect to server using url mongodb://hostfromenv:9216: no reachable servers" source="connection.go:84"
I0401 11:19:28.653] time="2019-04-01T11:18:59Z" level=error msg="Can't create mongo session to mongodb://hostfromenv:9216" source="mongodb_collector.go:200"
```

This matches up what the behaviour we'd expect according to [the docs](https://github.com/percona/mongodb_exporter#running).

> To define your own MongoDB URL, use environment variable `MONGODB_URI`. If set this variable takes precedence over `--mongodb.uri` flag.

File: stable/prometheus-mongodb-exporter/templates/deployment.yaml [hunk 2]
```diff
@@ -23,6 +23,11 @@ spec:
     spec:
       containers:
       - name: mongodb-exporter
+        env:
```
Comment (steven-sheehy) [inline stable/prometheus-mongodb-exporter/templates/deployment.yaml:27]: Add an if .Values.env above env then we should be good to go. Without it, env's empty value wouldn't be [] which can cause helm warnings when upgrading and changing value.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: stable/prometheus-mongodb-exporter/templates/deployment.yaml (hunk 2 above)
Comment (dashford) [reply stable/prometheus-mongodb-exporter/templates/deployment.yaml:27]: Super, changed here: https://github.com/helm/charts/pull/12615/commits/a602e3078e1aa47813935bd41ae5ed227bb8dc37

**Comment by steven-sheehy** [comment]
/lgtm

Thanks @dashford!

**Comment by dashford** [comment]
Thanks for the help @steven-sheehy :slightly_smiling_face:

**Pull request merged**